### PR TITLE
Custom Transformers & Fix OneOf fields

### PR DESCRIPTION
**Pull request by ysurtayev-bold**
- add custom (not generated) transformers for sub-messages
- add transformer.custom attribute to mark fields which require custom transformers
- limit oneOfDecl to the specific case it serves
- extend makefile with more commands

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,10 +1,10 @@
-VERSION=1.0.6-dev
+VERSION=1.0.7-dev
 BUILDTIME=$(shell date +"%Y-%m-%dT%T%z")
 LDFLAGS= -ldflags '-X github.com/bold-commerce/protoc-gen-struct-transformer/generator.version=$(VERSION) -X github.com/bold-commerce/protoc-gen-struct-transformer/generator.buildTime=$(BUILDTIME)'
 
 .PHONY: re-generate-example generate install build version setup
 
-re-generate-example:
+re-generate-example:
```
Comment (LiamPimlott) [inline Makefile]: trailing space

**Review by LiamPimlott — APPROVED** [review]
Did a first pass and nothing jumps out at me although I cant say I fully understand it all, might be helpful to have you explain a bit sometime

File: Makefile (hunk 1 above)
Comment (ysurtayev-bold) [reply Makefile]: Fixed. Thank you!

File: generator/oneof.go [hunk 2]
```diff
@@ -1,3 +1,6 @@
+// TODO:  This is a specific case of OneOf which is used by BoldCommerce and need to be removed from the plugin.
```
Comment (joshsimard) [inline generator/oneof.go]: Small nitpick, but I would say "...and need**s** to be removed from the plugin."

File: generator/template.go [hunk 3]
```diff
@@ -224,8 +224,11 @@ type Field struct {
 	// It true, field GoToProtoType and ProtoToGoType functions will be used
 	// with prefix.
 	UsePackage bool
-	OneofDecl  string
-	Opts       string
+	// The field has a value when it is used for the oneof migration from Int64 to String for the field
+	// TODO:  This is a specific case of OneOf which is used by BoldCommerce and need to be removed from the plugin.
```
Comment (joshsimard) [inline generator/template.go]: Same as above

**Review by joshsimard — COMMENTED** [review]
I didn't notice any glaring errors

File: generator/oneof.go (hunk 2 above)
Comment (ysurtayev-bold) [reply generator/oneof.go]: Fixed. Thank you!

File: generator/template.go (hunk 3 above)
Comment (ysurtayev-bold) [reply generator/template.go]: Fixed. Thanks

**Comment by gordcurrie** [comment]
Looks good to me, I have to admit I only went through the actual code changes, it looks like code coverage is not turned on for this repo, we'll fix that. If you're happy with the tests you have than that is good enough for me.

**Comment by lutherng** [comment]
LGTM

**Review by ytuz — COMMENTED** [review]
LGTM too!

**Review by joshsimard — APPROVED** [review]

**Pull request merged**